test/jdk/javax/swing/JTableHeader/TableHeaderBorderPositionTest.java

```diff
@@ -0,0 +1,129 @@
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Test to validate JTable header border vertical line is
+ * aligned with data grid lines (Metal L&F).
+ * @run main TableHeaderBorderPositionTest
+ */
+
+public class TableHeaderBorderPositionTest {
+    static JTable table;
+    private static final int WIDTH = 300;
+    private static final int HEIGHT = 150;
+    private static final double SCALE = 2.25;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        SwingUtilities.invokeAndWait(() -> {
+            try {
+                TableHeaderBorderPositionTest.Test();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    public static void Test() throws Exception {
+        int verticalLineCol;
+        int expectedRGB;
+        BufferedImage imgData;
+        BufferedImage imgHeader;
+        double w;
+        double h;
+        String[][] data = {
+                { "1", "1", "Green"},
+                { "2", "2", "Blue"}
+        };
+
+        String[] columnNames = { "Number", "Size", "Color"};
+
+        table = new JTable(data, columnNames);
+        table.setSize(WIDTH, HEIGHT);
+
+        final JTableHeader header = table.getTableHeader();
+        TableCellRenderer renderer = header.getDefaultRenderer();
+        header.setDefaultRenderer(renderer);
+        table.updateUI();
+
+        Dimension size = header.getPreferredSize();
+        header.setSize(size);
+        w = SCALE * size.width;
+        h = SCALE * size.height;
+        imgHeader = new BufferedImage((int)(w), (int)(h), BufferedImage.TYPE_INT_RGB);
+        Graphics2D g2d = imgHeader.createGraphics();
+        g2d.scale(SCALE, SCALE);
+        try {
+            header.paint(g2d);
+        } finally {
+            g2d.dispose();
+        }
+
+        w = SCALE * WIDTH;
+        h = SCALE * HEIGHT;
+
+        imgData = new BufferedImage((int)w, (int)h, BufferedImage.TYPE_INT_RGB);
+        g2d = imgData.createGraphics();
+        g2d.scale(SCALE, SCALE);
+        try {
+            table.paint(g2d);
+        } finally {
+            g2d.dispose();
+        }
+
+        verticalLineCol = (int)(table.getTableHeader().
+                getColumnModel().getColumn(0).getWidth() * SCALE) - 2;
+        expectedRGB = imgData.getRGB(verticalLineCol, 0);
+
+        for (int i = 0; i < imgHeader.getHeight(); i++) {
+            for (int j = verticalLineCol; j < verticalLineCol + 3; j++) {
+                if (expectedRGB != imgHeader.getRGB(j, i)) {
+                    throw new RuntimeException("Test Failed");
+                }
+            }
+        }
+
+        for (int i = 0; i < table.getRowCount() * table.getRowHeight() * SCALE; i++) {
+            for (int j = verticalLineCol; j < verticalLineCol + 3; j++) {
+                if (expectedRGB != imgData.getRGB(j, i)) {
```

Comment:
You probably can write the bufferedimage into png file for debugging help during fail case..